src/ngraph/runtime/cpu/pass/cpu_fusion.hpp

```diff
@@ -35,21 +35,40 @@ namespace ngraph
 class ngraph::runtime::cpu::pass::CPUFusion : public ngraph::pass::GraphRewrite
 {
 public:
```

Comment:
I don't like the 2^i enum with ors/ands. How about a regular enum and sets of values?
Also, `fusions` should be `Fusions`. If you make it `enum class Fusion` then `Fusion::differentiable` etc. will be easier to read than the C-like `DIFFERENTIABLE_FUSIONS`.

Reply:
We can definitely switch to `enum class`. Actually, my thinking to use regular enums was exactly the opposite: I didn't want introduce another namespace and `DIFFERENTIABLE_FUSIONS` seemed more readable to me than French's object + adjective :-) 

> I don't like the 2^i enum with ors/ands. How about a regular enum and sets of values?


This will probably result in code duplication since we won't be able to nest/reuse groups of fusions with CPUFusion as easily. It's pretty common even in higher-level languages like Java or c#. What I could do in case you really don't like this enum is convert it so something like this

```cpp
enum class Fusion
{
Regular,
Differentiable,
Invalid
}

std::bitset<Fusion::Invalid> fusion_groups;
```

then we could do ?

```cpp
fusion_groups.set(Fusion::Regular);
fusion_groups.set(Fusion::Differentiable);
```